### DISCUSSION TITLE
Automating DIA-NN Searches for timsTOF DIA Raw Files

### DISCUSSION THREAD

**Discussion by Raghothama1** [vdemichev/DiaNN#1039] [Q&A] [answered]
Dear DIA-NN Support Team,
We are currently analyzing a large number of timsTOF DIA raw files using DIA-NN. To streamline our workflow, we are interested in automating the searches so that they commence immediately after the mass spectrometry run is completed.
Could you please advise if there is a way to set up such automation within DIA-NN? Any guidance or documentation on how to achieve this would be greatly appreciated.
Thank you for your support.
Best regards,
Raghothama Chaerkady

**Comment by vdemichev** [comment]
Hi Raghothama,

Yes, I assume you want them search lib-free.
All you need:
- a spectral library - can be predicted or DIA-based
- run diann.exe with mass accuracies fixed to 15 ppm and the scan window fixed to the approximate number of data points per peak (usually, DIA methods are optimised for the values in the range 6-10, but it's perfectly OK to do it with 7 for almost any DIA run)

This will create a .quant file next to each .d folder. Also, it will produce an output report at whatever location you choose, with identifications. To keep things compact, I would suggest using --no-main-report and --no-stats, to only generate a .parquet output report.

Using the quant files produced, you can then create an analysis combining any number of raw files already processed, using --use-quant. If you've used a predicted library, will need to do the second step of MBR (i.e. run with --use-quant and --reanalyse) - but that's much quick than the search with a predicted library. If you've used an empirical library and MBR is not necessary - can just analyse with --use-quant and DIA-NN will not even need to access the raw files themselves, just the .quant files next to them. 

Best,
Vadim

**Reply by vdemichev** [reply, marked as answer]
Documentation is in the 'Flexible reanalysis' sub-section here https://github.com/vdemichev/DiaNN?tab=readme-ov-file#output
